// @mux/mux-node/util/jwt-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataTypeClaim = exports.isMuxJWTSignOptionsMultiple = exports.TypeToken = exports.TypeClaim = void 0;
var TypeClaim;
(function (TypeClaim) {
    TypeClaim["video"] = "v";
    TypeClaim["thumbnail"] = "t";
    TypeClaim["gif"] = "g";
    TypeClaim["storyboard"] = "s";
    TypeClaim["stats"] = "playback_id";
    TypeClaim["drm_license"] = "d";
})(TypeClaim = exports.TypeClaim || (exports.TypeClaim = {}));
var TypeToken;
(function (TypeToken) {
    TypeToken["video"] = "playback-token";
    TypeToken["thumbnail"] = "thumbnail-token";
    TypeToken["storyboard"] = "storyboard-token";
    TypeToken["drm_license"] = "drm-token";
    TypeToken["gif"] = "gif-token";
    TypeToken["stats"] = "stats-token";
})(TypeToken = exports.TypeToken || (exports.TypeToken = {}));
const isMuxJWTSignOptionsMultiple = (config) => Array.isArray(config.type);
exports.isMuxJWTSignOptionsMultiple = isMuxJWTSignOptionsMultiple;
var DataTypeClaim;
(function (DataTypeClaim) {
    DataTypeClaim["video"] = "video_id";
    DataTypeClaim["asset"] = "asset_id";
    DataTypeClaim["playback"] = "playback_id";
    DataTypeClaim["live_stream"] = "live_stream_id";
})(DataTypeClaim = exports.DataTypeClaim || (exports.DataTypeClaim = {}));
